### PR TITLE
[FX-2123] Add a formatted artwork count to ArtistSeries

### PR DESCRIPTION
**Pull request by anandaroop**
https://artsyproduct.atlassian.net/browse/FX-2123

### Problem

When an artist series has no for-sale works we don't want to simply show "0 available"

### Solution

- Use the count of all works (not just for-sale ones) as a fallback
- Return a consistently formatted string ("42 available" when there are for-sale works vs "42 works" when there are no for-sale works)

Originally I had this logic living in Force and Eigen, but for the sake of consistency and reusability it seemed sensible to have this in MP, since it is a common formatting concern across clients and surfaces. 

So this will be followed up by PRs to Force and Eigen to simply use the new `artworksCountMessage` field.

Screen Shot 2020-08-12 at 4 43 20 PM

Screen Shot 2020-08-12 at 4 43 26 PM

### CODE REVIEW

File: src/lib/stitching/gravity/__tests__/stitching.test.ts [hunk 1]
```diff
@@ -707,4 +707,55 @@ describe("gravity/stitching", () => {
...
+      const { resolvers } = await getGravityStitchedSchema()
+      const { artworksCountMessage } = resolvers.ArtistSeries
+
+      const value = await artworksCountMessage.resolve({
+        forSaleArtworksCount: 0,
+        artworksCount: 0,
+      })
+
+      expect(value).toEqual(null)
+    })
```
Comment (anandaroop) [inline src/lib/stitching/gravity/__tests__/stitching.test.ts:759]: This is not specified in any design but I figured returning a falsy value and not displaying a count at all was better than "0 works" — since that's what we were trying to avoid in the first place.

Comment (sweir27) [reply src/lib/stitching/gravity/__tests__/stitching.test.ts:759]: Ohh interesting. I probably would expect this to say `0 works`! But if that would cause us to then do some logic/checks on the presence of the "0 works" string, then this makes sense.

**Comment by sweir27** [comment]
Nice tests! 

**thought (non-blocking)**: I have slight reservations against making the string `null` if there are `0 works` but it's not a strong opinion. Feel free to merge as-is!

File: src/lib/stitching/gravity/__tests__/stitching.test.ts (hunk 1 above)
Comment (anandaroop) [reply src/lib/stitching/gravity/__tests__/stitching.test.ts:759]: Cool, so I've put in a question to Nicole about whether we should show a "0 works" message or nothing at all for this edgy edge case — and will let that guide the decision 😎.

Comment (sweir27) [reply src/lib/stitching/gravity/__tests__/stitching.test.ts:759]: Sounds good!

File: src/lib/stitching/gravity/stitching.ts [hunk 2]
```diff
@@ -197,6 +202,23 @@ export const gravityStitchingEnvironment = (
...
+        artworksCountMessage: {
+          fragment: gql`
+            ... on ArtistSeries {
+              forSaleArtworksCount
+              artworksCount
+            }
+          `,
+          resolve: async ({ forSaleArtworksCount, artworksCount }) => {
+            const artworksCountMessage = forSaleArtworksCount
+              ? `${forSaleArtworksCount} available`
```
Comment (williardx) [inline src/lib/stitching/gravity/stitching.ts:214]: **suggestion (non-blocking):** Reformat as explicit `if/else` statements since the nested ternaries make it difficult to quickly grasp what's going on. Non-blocking mostly because it's still clear, just requires a little more effort.

Comment (anandaroop) [reply src/lib/stitching/gravity/stitching.ts:214]: Good call

File: src/lib/stitching/gravity/__tests__/stitching.test.ts (hunk 1 above)
Comment (anandaroop) [reply src/lib/stitching/gravity/__tests__/stitching.test.ts:759]: Confirmed that we want to show "0 works" — commit incoming

Comment (anandaroop) [reply src/lib/stitching/gravity/__tests__/stitching.test.ts]: Fixed in https://github.com/artsy/metaphysics/pull/2617/commits/668c3733457c6075c765deeed9c6ff22754418ca#diff-e9a6ef1ded9b9a8e03c9320518a09a23R749

File: src/lib/stitching/gravity/stitching.ts (hunk 2 above)
Comment (anandaroop) [reply src/lib/stitching/gravity/stitching.ts]: Fixed in https://github.com/artsy/metaphysics/commit/668c3733457c6075c765deeed9c6ff22754418ca#diff-573ae00d2f637334b78f6df4edf48571R215-R221

**Comment by anandaroop** [comment]
Thanks for the comments — both points addressed. This should be g2g 😄
